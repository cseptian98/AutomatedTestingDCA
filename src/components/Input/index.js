import React, {useState} from 'react';
import {View, TextInput, TouchableOpacity, StyleSheet} from 'react-native';
import axiosConfig from '../../api/BaseConfig';

const InputTodo = () => {
  const [title, setTitle] = useState('');

  const doCreate = () => {
    const onSuccess = ({data}) => {
      console.log('debug success', data);
    };

    const onFailure = error => {
      console.log('debug error', error.response.data);
    };

    axiosConfig.post('api/TodoLists', {title}).then(onSuccess).catch(onFailure);
  }

  return (
    <View style={styles.container}>
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Add Todo List..."
          placeholderTextColor="#000"
          onChangeText={title => setTitle(title)}
        />
      </View>
      <TouchableOpacity style={styles.button} onPress={doCreate}>
        {/* ICON */}
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginTop: 20,
  },
  inputView: {
    borderRadius: 10,
    flexDirection: 'row',
  },
  textInput: {
    width: 300,
    padding: 10,
    marginLeft: 10,
    marginRight: 10,
    marginBottom: 20,
    borderRadius: 10,
    backgroundColor: '#FFD500',
  },
  button: {
    width: 50,
    marginBottom: 20,
    borderRadius: 50,
    backgroundColor: '#FFD500',
  },
});

export default InputTodo;
