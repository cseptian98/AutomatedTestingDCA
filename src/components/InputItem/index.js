import React, {useState} from 'react';
import {View, TextInput, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import axiosConfig from 'api/BaseConfig';
import styles from './InputItems.styles';

const InputItem = () => {
  const listId = 1;
  const [title, setTitle] = useState('');

  const createNewItem = () => {
    const onSuccess = ({data}) => {
      console.log('debug success', data);
      setTitle('');
    };

    const onFailure = error => {
      console.log('debug error', error);
    };

    axiosConfig
      .post('api/TodoItems', {listId, title})
      .then(onSuccess)
      .catch(onFailure);
  };

  return (
    <View style={styles.container}>
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Add Todo Item..."
          placeholderTextColor="#000"
          onChangeText={title => setTitle(title)}
        />
      </View>
      <TouchableOpacity style={styles.button} onPress={createNewItem}>
        <Icon name="add" size={24} />
      </TouchableOpacity>
    </View>
  );
};

export default InputItem;
