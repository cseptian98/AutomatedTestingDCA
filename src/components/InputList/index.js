import React, {useState} from 'react'
import {View, TextInput, TouchableOpacity, Image} from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'
import styles from './InputList.styles'
import axiosConfig from 'api/BaseConfig'

const InputList = ({createList}) => {
  const [title, setTitle] = useState('')

  // const createNewList = () => {
  //   const onSuccess = ({data}) => {
  //     console.log('debug success', data)
  //     setTitle('')
  //   }

  //   const onFailure = error => {
  //     console.log('debug error', error)
  //   }

  //   axiosConfig.post('api/TodoLists', {title}).then(onSuccess).catch(onFailure)
  // }

  return (
    <View style={styles.container}>
      <View>
        <Image
          style={styles.image}
          source={require('assets/images/list.png')}
        />
      </View>
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Add Todo List..."
          placeholderTextColor="#FFF"
          onChangeText={title => setTitle(title)}
        />
        <TouchableOpacity
          style={styles.button}
          onPress={() => createList(title)}>
          <Icon name="add" size={30} color="#FFF" />
        </TouchableOpacity>
      </View>
    </View>
  )
}

export default InputList
