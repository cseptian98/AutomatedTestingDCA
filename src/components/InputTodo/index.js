import React from 'react';
import {View, TextInput, TouchableOpacity, StyleSheet} from 'react-native';

const InputTodo = () => {
  return (
    <View style={styles.container}>
      <View style={styles.inputView}>
        <TextInput
          style={styles.textInput}
          placeholder="Add Todo List..."
          placeholderTextColor="#000"
        />
      </View>
      <TouchableOpacity style={styles.button}>
        {/* ICON */}
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  inputView: {
    borderRadius: 10,
    flexDirection: 'row',
  },
  textInput: {
    width: 300,
    padding: 10,
    marginLeft: 10,
    marginRight: 10,
    marginBottom: 20,
    borderRadius: 10,
    backgroundColor: '#FFD500',
  },
  button: {
    width: 50,
    marginBottom: 20,
    borderRadius: 50,
    backgroundColor: '#FFD500',
  },
});

export default InputTodo;
