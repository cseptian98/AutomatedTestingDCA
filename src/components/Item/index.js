import React from 'react'
import {TouchableOpacity, Text} from 'react-native'
import styles from './Item.styles'

const Item = ({title, onDelete}) => (
  <TouchableOpacity style={styles.item} onPress={onDelete}>
    <Text style={styles.title}>{title}</Text>
  </TouchableOpacity>
)

export default Item
