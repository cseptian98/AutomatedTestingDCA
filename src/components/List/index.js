import React from 'react';
import {TouchableOpacity, Text, View} from 'react-native';
import styles from './List.styles';

const List = ({title, onDelete}) => (
  <TouchableOpacity style={styles.item} onPress={onDelete}>
    <Text style={styles.title}>{title}</Text>
  </TouchableOpacity>
);

export default List;
