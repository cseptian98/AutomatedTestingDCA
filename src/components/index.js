import FormUpdateItem from './TodoItems/FormUpdateItem/FormUpdateItem'
import TodoLists from './TodoLists/TodoLists'
import TodoItems from './TodoItems/TodoItems'
import User from './User/User'

export {FormUpdateItem, TodoLists, TodoItems, User}