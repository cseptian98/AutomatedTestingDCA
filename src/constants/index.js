export const TEST_ID_IMAGE_LOGIN = 'TEST_ID_IMAGE_LOGIN'
export const TEST_ID_EMAIL_LOGIN = 'TEST_ID_EMAIL_LOGIN'
export const TEST_ID_PASSWORD_LOGIN = 'TEST_ID_PASSWORD_LOGIN'
export const TEST_ID_BUTTON_LOGIN = 'TEST_ID_BUTTON_LOGIN'
export const TEST_ID_BUTTON_CREATE_ACCOUNT = 'TEST_ID_CREATE_ACCOUNT'
export const TEST_ID_IMAGE_REGISTER = 'TEST_ID_IMAGE_REGISTER'
export const TEST_ID_EMAIL_REGISTER = 'TEST_ID_EMAIL_REGISTER'
export const TEST_ID_PASSWORD_REGISTER = 'TEST_ID_PASSWORD_REGISTER'
export const TEST_ID_CONFIRMATION_PASSWORD = 'TEST_ID_CONFIRMATION_PASSWORD'
export const TEST_ID_BUTTON_REGISTER = 'TEST_ID_BUTTON_REGISTER'
export const TEST_ID_TODOLIST = 'TEST_ID_TODOLIST'
export const TEST_ID_IMAGE_TODOLIST = 'TEST_ID_IMAGE_TODOLIST'
export const TEST_ID_TEXT_INPUT_LIST = 'TEST_ID_TEXT_INPUT_LIST'
export const TEST_ID_BUTTON_SUBMIT_LIST = 'TEST_ID_BUTTON_SUBMIT_LIST'
export const TEST_ID_TITLE_TODOLIST = 'TEST_ID_TITLE_TODOLIST'
export const TEST_ID_BUTTON_UPDATE_LIST = 'TEST_ID_BUTTON_UPDATE_LIST'
export const TEST_ID_BUTTON_DELETE_LIST = 'TEST_ID_BUTTON_DELETE_LIST'
export const TEST_ID_TODOITEM = 'TEST_ID_TODOITEM'
export const TEST_ID_IMAGE_TODOITEM = 'TEST_ID_IMAGE_TODOITEM'
export const TEST_ID_TEXT_INPUT_ITEM = 'TEST_ID_TEXT_INPUT_ITEM'
export const TEST_ID_BUTTON_SUBMIT_ITEM = 'TEST_ID_BUTTON_SUBMIT_ITEM'
export const TEST_ID_TITLE_TODOITEM = 'TEST_ID_TITLE_TODOITEM'
export const TEST_ID_BUTTON_UPDATE_ITEM = 'TEST_ID_BUTTON_UPDATE_ITEM'
export const TEST_ID_BUTTON_DELETE_ITEM = 'TEST_ID_BUTTON_DELETE_ITEM'
export const TEST_ID_IMAGE_USER = 'TEST_ID_IMAGE_USER'
export const TEST_ID_BUTTON_LOGOUT = 'TEST_ID_BUTTON_LOGOUT'
export const TEST_ID_IMAGE_UPDATE_ITEM = 'TEST_ID_IMAGE_UPDATE_ITEM'
export const TEST_ID_NOTES_ITEM = 'TEST_ID_NOTES_ITEM'
export const TEST_ID_BUTTON_UPDATE_DETAIL = 'TEST_ID_BUTTON_UPDATE_DETAIL'
