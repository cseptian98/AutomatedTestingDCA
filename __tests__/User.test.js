import React from 'react'
import {fireEvent, render} from '@testing-library/react-native'
import {User} from 'components'
import {TEST_ID_BUTTON_LOGOUT, TEST_ID_IMAGE_USER} from 'constants'

let componentUser

describe('User Screen', () => {
  beforeEach(() => {
    componentUser = render(<User />)

    buttonLogin = componentUser.getByTestId(TEST_ID_BUTTON_LOGOUT)
  })

  it('should show user screen', () => {
    expect(componentUser.getByTestId(TEST_ID_IMAGE_USER)).toBeTruthy()
  })
})
