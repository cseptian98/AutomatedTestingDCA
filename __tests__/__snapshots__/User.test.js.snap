// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/images/profil.png",
      }
    }
    style={
      Object {
        "height": 200,
        "marginBottom": 30,
        "marginTop": 30,
        "width": 240,
      }
    }
    testID="TEST_ID_IMAGE_USER"
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#B71C1C",
        "borderRadius": 25,
        "color": "#000",
        "height": 50,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "width": "80%",
      }
    }
    testID="TEST_ID_BUTTON_LOGOUT"
  >
    <Text
      style={
        Object {
          "color": "#FFF",
        }
      }
    >
      Logout
    </Text>
  </View>
</View>
`;
